judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/AbstractTest.java

```diff
@@ -0,0 +1,538 @@
+
+        List<J> jMultiRelation = List.of(
+                jDao.create(J.builder().build()),
+                jDao.create(J.builder().build()),
+                jDao.create(J.builder().build())
+        );
+        List<I> iMultiRelation = jMultiRelation.stream().map(l -> JtoI(l)).collect(Collectors.toList());
+
+        H h = hDao.create(H.builder()
+                        .withCompositionIOnHSingle(JtoI(jDao.create(J.builder().build())))
+                        .withCompositionIOnHSingleRequired(JtoI(jDao.create(J.builder().build())))
+                        .withCompositionIOnHMulti(List.of(
+                                jDao.create(J.builder().withNameI("V").build()),
+                                jDao.create(J.builder().withNameI("V").build()),
+                                jDao.create(J.builder().withNameI("V").build())
+                        ).stream().map(l -> JtoI(l)).collect(Collectors.toList()))
+                        .build(),
+                HAttachedRelationsForCreate.builder()
+                        .withRelationIOnHSingle(iOptionalRelation)
+                        .withRelationIOnHSingleRequired(iRequiredRelation)
+                        .withRelationIOnHMulti(iMultiRelation)
+                        .build()
+        );
+
+        //Check j and i instance ar equal because i is live with j only
+        assertEquals(iDao.getAll().size(), jDao.getAll().size());
+
+        //update
+
+        h.getCompositionIOnHSingle().get().setNameI("CompositionSingleChanged");
+        h.getCompositionIOnHSingleRequired().setNameI("CompositionSingleRequiredChanged");
+        h.getCompositionIOnHMulti().forEach(i -> i.setNameI("MultiCompositionNameChanged"));
+
+        iOptionalRelation.setNameI("iOptionalRelationNameChanged");
+        iRequiredRelation.setNameI("iRequiredRelationNameChanged");
+        iMultiRelation.forEach(i -> i.setNameI("iMultiRelationNameChanged"));
+
+
+        iDao.update(h.getCompositionIOnHSingle().get());
+        iDao.update(h.getCompositionIOnHSingleRequired());
+        h.getCompositionIOnHMulti().stream().map(i -> iDao.update(i));
+
+        iOptionalRelation = iDao.update(iOptionalRelation);
+        iRequiredRelation = iDao.update(iRequiredRelation);
+        iMultiRelation = iMultiRelation.stream().map(i -> iDao.update(i)).collect(Collectors.toList());
+
+        h = hDao.getById(h.identifier()).orElseThrow();
+
+        //Check update
+        assertEquals(Optional.of("CompositionSingleChanged"), h.getCompositionIOnHSingle().orElseThrow().getNameI());
+        assertEquals(Optional.of("CompositionSingleRequiredChanged"), h.getCompositionIOnHSingleRequired().getNameI());
+
+        //TODO JNG-4813
+        //h.getCompositionIOnHMulti().stream().forEach( i -> assertEquals(Optional.of("MultiCompositionNameChanged"),i.getNameI()));
+        //hDao.queryCompositionIOnHMulti.execute().stream().forEach( i -> assertEquals(Optional.of("iMultiRelationNameChanged"),i.getNameI()));
```

Comment:
Sure query is not available but `query[RELATION].execute()` is the same as `get[RELATION]()` (where said relation is a collection)

Please update test case

(query method can be tested later)

Reply:
[6f85e57](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/165/commits/6f85e57884996176f05ef4554ae4669dee9e50a2)